### PR TITLE
Move aws helper functions to seperate pkg/aws

### PR DESCRIPTION
**Pull request by PrasadG193**
## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [x] :rainbow: Refactoring (no functional changes, no api changes)
- [ ] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [ ] :sunflower: Feature
- [ ] :world_map: Documentation
- [ ] :robot: Test

## Issues

- #XXX

## Test Plan

- [ ] :muscle: Manual
- [ ] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

File: pkg/aws/aws.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+const (
+	maxRetries = 10
+)
+
+// GetConfigFromProfile extracts AWS creds from profile
+func GetConfigFromProfile(profile *param.Profile) (map[string]string, error) {
+	config := make(map[string]string)
+	if profile.Location.Type != crv1alpha1.LocationTypeS3Compliant {
+		return nil, errors.New("Not a S3Compliant location")
+	}
```
Comment (PrasadG193) [inline pkg/aws/aws.go]: Remove this check

File: pkg/aws/aws.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+		config[awsconfig.SecretAccessKey] = string(profile.Credential.Secret.Data[secrets.AWSSecretAccessKey])
+		config[awsconfig.ConfigRole] = string(profile.Credential.Secret.Data[secrets.ConfigRole])
+		config[awsconfig.SessionToken] = string(profile.Credential.Secret.Data[secrets.AWSSessionToken])
+	}
+	config[awsconfig.ConfigRegion] = profile.Location.Region
+	return config, nil
+}
+
+// NewConfigWithSession creates aws.Config with session
+func NewConfigWithSession(ctx context.Context, config map[string]string) (*aws.Config, *session.Session, error) {
```
Comment (vkamra) [inline pkg/aws/aws.go]: Should we return a `session` from this method? Why not just return `NewConfig` and let the service clients (e.g. NewEC2Client) create a session.

File: pkg/aws/rds.go [hunk 3]
```diff
@@ -0,0 +1,77 @@
+// Copyright 2019 The Kanister Authors.
```
Comment (vkamra) [inline pkg/aws/rds.go]: I think we should organize the `aws` pkg - move `rds`, `ec2`, `config` into sub-packages.

Also `rds.NewClient()` works better than `aws.NewRDSClient()`

Comment (PrasadG193) [reply pkg/aws/rds.go]: Makes sense. Will do the changes

File: pkg/aws/aws.go (hunk 2 above)
Comment (PrasadG193) [reply pkg/aws/aws.go]: Sounds good

File: pkg/aws/aws.go [hunk 4]
```diff
@@ -39,8 +43,41 @@ const (
...
+		config[AccessKeyID] = string(profile.Credential.Secret.Data[secrets.AWSAccessKeyID])
+		config[SecretAccessKey] = string(profile.Credential.Secret.Data[secrets.AWSSecretAccessKey])
+		config[ConfigRole] = string(profile.Credential.Secret.Data[secrets.ConfigRole])
+		config[SessionToken] = string(profile.Credential.Secret.Data[secrets.AWSSessionToken])
+	}
+	config[ConfigRegion] = profile.Location.Region
+	return GetConfig(ctx, config)
+}
+
+//// NewConfigWithSession creates aws.Config with session
```
Comment (vkamra) [inline pkg/aws/aws.go]: Remove commented code

**Comment by vkamra** [comment]
This LGTM. I'm going to address the review comment and the CI failure so that we can merge

File: pkg/objectstore/objectstore_test.go [hunk 5]
```diff
@@ -31,7 +31,7 @@ import (
 	"google.golang.org/api/compute/v1"
 	. "gopkg.in/check.v1"
 
-	"github.com/kanisterio/kanister/pkg/config/aws"
+	"github.com/kanisterio/kanister/pkg/aws"
```
Comment (SupriyaKasten) [inline pkg/objectstore/objectstore_test.go]:
```suggestion
	"github.com/kanisterio/kanister/pkg/aws"
	awsrole "github.com/kanisterio/kanister/pkg/aws/role"
```

Comment (SupriyaKasten) [reply pkg/objectstore/objectstore_test.go]: Please update line 461 to `awsrole.Switch( ....`

File: pkg/objectstore/objectstore_test.go [hunk 6]
```diff
@@ -458,7 +459,7 @@ func getSecret(c *C, ctx context.Context, osType ProviderType) *Secret {
 		awsSecretAccessKey := os.Getenv("AWS_SECRET_ACCESS_KEY")
 		var awsSessionToken string
 		if role, ok := os.LookupEnv(aws.ConfigRole); ok {
-			creds, err := aws.SwitchRole(ctx, awsAccessKeyID, awsSecretAccessKey, role, assumeRoleDuration)
+			creds, err := awsrole.SwitchRole(ctx, awsAccessKeyID, awsSecretAccessKey, role, assumeRoleDuration)
```
Comment (SupriyaKasten) [inline pkg/objectstore/objectstore_test.go]:
```suggestion
			creds, err := awsrole.Switch(ctx, awsAccessKeyID, awsSecretAccessKey, role, assumeRoleDuration)
```

**Pull request merged**